src/perf/bin/drvmain.cpp

```diff
@@ -0,0 +1,563 @@
+
+_No_competing_thread_
+INITCODE
+NTSTATUS
+QuicPerfCtlInitialize(
+    _In_ WDFDRIVER Driver
+    );
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicPerfCtlUninitialize(
+        void
+    );
+
+void* __cdecl operator new (size_t Size) {
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED, Size, QUIC_PERF_TAG);
+}
+
+void __cdecl operator delete (_In_opt_ void* Mem) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+void __cdecl operator delete (_In_opt_ void* Mem, _In_opt_ size_t) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+void* __cdecl operator new[](size_t Size) {
+    return ExAllocatePool2(POOL_FLAG_NON_PAGED, Size, QUIC_PERF_TAG);
+}
+
+void __cdecl operator delete[](_In_opt_ void* Mem) {
+    if (Mem != nullptr) {
+        ExFreePoolWithTag(Mem, QUIC_PERF_TAG);
+    }
+}
+
+extern "C" _IRQL_requires_max_(PASSIVE_LEVEL) void QuicTraceRundown(void) { }
+
+extern "C"
+INITCODE
+_Function_class_(DRIVER_INITIALIZE)
+_IRQL_requires_same_
+_IRQL_requires_(PASSIVE_LEVEL)
+NTSTATUS
+DriverEntry(
+    _In_ PDRIVER_OBJECT DriverObject,
+    _In_ PUNICODE_STRING RegistryPath
+    )
+{
+    NTSTATUS Status;
+    WDF_DRIVER_CONFIG Config;
+    WDFDRIVER Driver;
+    BOOLEAN PlatformInitialized = FALSE;
+
+    QuicPlatformSystemLoad(DriverObject, RegistryPath);
+
+    Status = QuicPlatformInitialize();
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "QuicPlatformInitialize failed");
+        goto Error;
+    }
+    PlatformInitialized = TRUE;
+
+    //
+    // Create the WdfDriver Object
+    //
+    WDF_DRIVER_CONFIG_INIT(&Config, NULL);
+    Config.EvtDriverUnload = QuicPerfDriverUnload;
+    Config.DriverInitFlags = WdfDriverInitNonPnpDriver;
+    Config.DriverPoolTag = QUIC_PERF_TAG;
+
+    Status =
+        WdfDriverCreate(
+            DriverObject,
+            RegistryPath,
+            WDF_NO_OBJECT_ATTRIBUTES,
+            &Config,
+            &Driver);
+    if (!NT_SUCCESS(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "WdfDriverCreate failed");
+        goto Error;
+    }
+
+    //
+    // Initialize the device control interface.
+    //
+    Status = QuicPerfCtlInitialize(Driver);
+    if (!NT_SUCCESS(Status)) {
+        goto Error;
+    }
+
+    QuicTraceLogInfo(
+        PerfDriverStarted,
+        "[perf] Started");
+
+Error:
+
+    if (!NT_SUCCESS(Status)) {
+        if (PlatformInitialized) {
+            QuicPlatformUninitialize();
+        }
+        QuicPlatformSystemUnload();
+    }
+
+    return Status;
+}
+
+_Function_class_(EVT_WDF_DRIVER_UNLOAD)
+_IRQL_requires_same_
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicPerfDriverUnload(
+    _In_ WDFDRIVER Driver
+    )
+{
+    UNREFERENCED_PARAMETER(Driver);
```

Comment:
Instead of this, just comment out the Driver variable name `/* Driver */` above.